test/filter.test.js

```diff
@@ -5,7 +5,7 @@
 'use strict';
 var assert = require('assert');
 var should = require('should');
-var filter = require('../');
+var filter = require('../index.js').filterNodes;
```

Comment:
I think `require('../').filterNode` is better, no need to rely on a specific `main` file.